libshortfin/src/CMakeLists.txt

```diff
@@ -29,4 +29,6 @@ shortfin_public_library(
     ${_INIT_INTERNAL_DEPS}
 )
 
-set_target_properties(shortfin PROPERTIES VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR} SOVERSION ${SOVERSION})
+if(SHORTFIN_BUILD_DYNAMIC)
+  set_target_properties(shortfin PROPERTIES VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR} SOVERSION ${SOVERSION})
```

Comment:
Do you  think we should set this in shortfin_public_library function defination instead?

Reply:
I can't remember to have seen a version set on a static library, so I think we don't need to it to the function. But we can later on if we want to set the version number there.

Reply:
Sorry, maybe i miscommunicated. I meant maybe we can set it here: https://github.com/nod-ai/sharktank/blob/main/libshortfin/build_tools/cmake/shortfin_library.cmake#L39

Reply:
Great suggestion! I updated and renamed the patch accordingly.